packages/webapp/src/components/Finances/AddExpense/ExpenseItemsForType.jsx

```diff
@@ -13,22 +13,42 @@
  *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
  */
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import { useTranslation } from 'react-i18next';
 import { useFieldArray } from 'react-hook-form';
 import { AddLink, Main } from '../../Typography';
 import ExpenseItemInputs from './ExpenseItemInputs';
 import { getInputErrors } from '../../Form/Input';
 import { NOTE, VALUE, EXPENSE_DETAIL } from './constants';
+import { selectedExpenseSelector } from '../../../containers/Finances/selectors';
+import { setSelectedExpenseTypes } from '../../../containers/Finances/actions';
 import styles from './styles.module.scss';
 
-export default function ExpenseItemsForType({ type, register, control, setValue, errors }) {
+export default function ExpenseItemsForType({ type, register, control, getValues, errors }) {
   const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const selectedExpense = useSelector(selectedExpenseSelector);
+
   const { fields, append, remove } = useFieldArray({
     control,
     name: `${EXPENSE_DETAIL}.${type.id}`,
   });
 
+  // Unselect the type when all items are removed, select the type when the first item is added.
+  // If the user removes items for the type, it will be unselected in the previous page when going back.
+  const handleSelectedExpenseTypes = (actionType) => {
```

Comment:
Just a minor point but I would have a slight preference to call this function parameter something else because the association between "action type" and Redux is so strong!

Reply:
Thank you Joyce! I struggled with coming up with the name actually... how would you call it?

Reply:
So minor maybe it's pointless, but just `operationType` perhaps?

Reply:
I like it, thank you!!